File: sample-chat-obj-c/sample-chat/Components/DialogsScreen/Views/TitleView.m
//
//  TitleView.m
//  sample-chat
//
//  Created by Injoit on 1/31/20.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "TitleView.h"

@implementation TitleView
#pragma mark - Utility
- (void)setupTitleViewWithTitle:(NSString *)title subTitle:(NSString *)subTitle {
    NSString *titleString = [NSString stringWithFormat:@"%@\n%@", title, subTitle ];
    NSMutableAttributedString *attrString = [[NSMutableAttributedString alloc] initWithString:titleString];
    
    NSRange titleRange = [titleString rangeOfString:title];
    UIFont *fontTitle = [UIFont systemFontOfSize:17.0f weight:UIFontWeightBold];
    NSDictionary *attributesTitle = @{ NSForegroundColorAttributeName:UIColor.whiteColor,
                                  NSFontAttributeName:fontTitle};
    [attrString addAttributes:attributesTitle range:titleRange];
    
    NSRange subTitleRange = [titleString rangeOfString:subTitle];
    UIFont *fontSubTitle = [UIFont systemFontOfSize:13.0f];
    NSDictionary *attributesSubTitle = @{ NSForegroundColorAttributeName:UIColor.whiteColor,
                                  NSFontAttributeName:fontSubTitle};
    [attrString addAttributes:attributesSubTitle range:subTitleRange];
    
    self.numberOfLines = 2;
    self.attributedText = attrString;
    self.textAlignment = NSTextAlignmentCenter;
    
    CGRect frame;
    frame = self.frame;
    frame.size.width = 200.0f;
    self.frame = frame;
}


@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/Classes/Views/TitleView.swift
//
//  TitleView.swift
//  sample-conference-videochat-swift
//
//  Created by Injoit on 10/10/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class TitleView: UILabel {
    func setupTitleView(title: String, subTitle: String) {
    
        let titleString = title + "\n" + subTitle
        
        let attrString = NSMutableAttributedString(string: titleString)
        
        let titleRange: NSRange = (titleString as NSString).range(of: title)
        attrString.addAttribute(.font, value: UIFont.boldSystemFont(ofSize: 17.0), range: titleRange)
        attrString.addAttribute(.foregroundColor, value: UIColor.white, range: titleRange)
        
        let numberChatsRange: NSRange = (titleString as NSString).range(of: subTitle)
        attrString.addAttribute(.font, value: UIFont.systemFont(ofSize: 13.0), range: numberChatsRange)
        attrString.addAttribute(.foregroundColor, value: UIColor.white.withAlphaComponent(0.6), range: numberChatsRange)
        
        numberOfLines = 2
        attributedText = attrString
        textAlignment = .center
        bounds.size.width = 200.0
    }
}
